### ISSUE TITLE
Uhuru V1.4 Beta

### ISSUE THREAD

**Issue by ken-de-moor** [threefoldtech/home#1260]
For the V1.4 Beta we continue tackling the UI/UX issues and bugs. This version we are also adding mobile support with a full scaleable layout. A full rewrite of the entire backend stack is included in a huge 54-point ticket.

https://github.com/threefoldtech/twin_aydo/projects/7

**Comment by sasha-astiadi** [comment]
Is this duplicate of #1248 ? please confirm /close edit @ken-de-moor

**Comment by ken-de-moor** [comment]
Not a duplicate, this is just the same version with the full rewrite included and merged. It's a huge change.

**Comment by ken-de-moor** [comment]
Following things fixe :

AUTH
 - [x] Fix auth guard
 - [x] Create new contact guard for external routes

USER
 - [x] Logout
 - [x] Change profile picture
 - [x] Update status

FLOW

- [x]  post doesn't show images.
- [x]  can't see posts from friend (even with reloading page).
- [x]  can't change avatar (removes the default one and leaves blank), also breaks everything.
- [x]  refreshing page after changing profile description resets it to default.
- [x] Delete user dialog 'cancel' button doesn't work and doesn't get closed properly
- [x] Deleting user still shows new posts from that user
- [x] Only show posts of the contact if you accept the request
- [x] Only show posts of the contact if he accepted your request
- [x] Stop showing the posts of the contact if he removes you as a 'friend'
- [x] Remove contact (from Redis) if user declines it
- [x]  share post doesn't work.
- [x]  hovering over post avatar gives error (should show profile info), And breaks everything!
- [x]  clicking on name does nothing (shows cursor pointer).

WHISPER

- [x] System messages always show at bottom of chat.
- [x] Update read messages
- [x]  Changing profile description doesn't change in chat info.
- [x] Promoting/demoting users not implemented yet
- [x]  Add chat with yourself shows 'delete group' in chat info.
- [x]  only shows video chat has started after refreshing page.
- [x]  message from input doesn't get saved when switching between navigation.
- [x]  Deleting messages only works if the other user refreshes his page.
- [x] Voice message doesn't get saved in own chat
- [x]  deleting voice message doesn't get send to other users.
- [x]  Replying on message doesn't get send to other user.
- [x]  doesn't always scroll down with new messages.
- [x]  editing messages doesn't work.
- [x]  Able to add yourself whe/chatn creating a group(makes you appear double in the group), And not able to leave the group afterwards.
- [x]  restarting docker removes all storage.
- [x]  Images don't get send to the other users in group chat (only for yourself).
- [x]  After deleting user from group he doesn't show up in 'add dialog'. And update message only shows after reloading page.
- [x]  Leaving group as non admin shows wrong 'leave dialog', makes impossible to leave group.
- [x]  Adding user to group chat after group is created is bugged.
     - [x]  not updating members.
     - [x]  doesn't close 'add dialog' for admin.
     - [x]  system messages don't show up until page is refreshed.
     - [x]  Added user can't leave the group.
     - [x]  Added user doesn't receive group messages but his own messages do get send to the others

QUANTUM
	
- [x] Edit shares dialog is always empty
- [x] Delete share does not work
- [x] Can not download files with `(`, `)`, or `+` in their names
- [x] Remove `(` and `)` from special characters
- [x] Make maximum file size bigger
- [x] Renaming file breaks the share
- [x] Moving file breaks the share
- [x] Bug: All published post images also show in quantum storage (should be saved in different folder)
- [x]  sharing files doesn't work
- [x]  opening documents doesn't work

**Comment by ken-de-moor** [comment]
Working on:

FLOW
 - [x] Clicking on +6 photo does nothing on post(I need to click on 'show all images' -> not so visible) 
 - [x] Clicking on comments or reply should autofocus on input field
 - [x] Can't click on an image in share post dialog (gives console error: _ctx.openImagePreview is not a function)
 - [x] Show correct avatar in sharePostDialog for group chats
 - [x] Sharing post with group also shares it with other chats
 - [x] Hovering over user name in comments does nothing but shows 'cursor pointer' and text underline

WHISPER
 - [x] Deleting chat also removes the contact from contact list. Contact list in the frontend should use 'contacts' from redis instead of chat list.
 - [x] Deleting chat shows 'leaving group' should be 'Deleting chat'
 - [x] Blocking user breaks whisper layout
 - [x] Blocked contact gets unblocked after refreshing page
 - [x] Should be able to add user to group by clicking on the whole list item instead of only the radio button
 - [x] Add contact dialog showed up 2 times over each other
 - [x] Adding a chat when you are still contacts with each other should not show the message 'request has been sent'
 - [x] New group chat doesn't shows as 'unread chat' for the added user if there are no messages yet
 - [x] Can't leave group as admin if there are still other users in it
	
QUANTUM
 - [ ] Onlyoffice shows warning and error (download failed)
 - [ ] The input field in create folder dialog should get autofocus
 - [ ] Breadcrumbs are wrong
 - [ ] Sharing file in group chat only gets send to admin and the person who shared the file
 - [ ] Can't view shared file in chat
 - [ ] Removing share doesn't delete the message in whisper (for group chats and single chat)
 - [ ] Deleting file doesn't remove all shares and messages in chat
 - [ ] Sharing the same file multiple times adds a new message every time in your own chat. It should delete the existing message and add a new message at the bottom of the chat.
 - [ ] Deleting 'shared file' message shows 'NaN undefined' after refreshing -> Deleted messages need to be removed instead of showing 'message deleted' after refreshing
 - [ ] Can't go into a folder or view files on mobile (only select)
 - [ ] Can't add files with the plus sign on mobile (bottom right)

GLASS
 - [ ] Missed messages notification is gone when on glass page
 - [ ] Going from glass to other page doesn't close the navigation when clicked on (have to manually close it)

**Comment by sasha-astiadi** [comment]
3.6.1

**Comment by xmonader** [comment]
this can't be 3.6.1, 3.6.1 has only dashboard for 3.6.0 release. @sasha-astiadi

**Comment by ken-de-moor** [comment]
Done and live
